packages/shared/src/porter.ts

```diff
@@ -120,10 +133,47 @@ export type TacoDecryptResult = {
 };
 
 export class PorterClient {
-  readonly porterUrl: URL;
+  readonly porterUrls: URL[];
+
+  constructor(porterUris: string | string[]) {
+    if (porterUris instanceof Array) {
+      this.porterUrls = porterUris.map((uri) => new URL(uri));
+    } else {
+      this.porterUrls = [new URL(porterUris)];
+    }
+  }
 
-  constructor(porterUri: string) {
-    this.porterUrl = new URL(porterUri);
+  protected async tryAndGet<T, D>(
```

Comment:
You can make this even more generic by using `axios.request` and passing in `GET` and `POST` methods

Reply:
thanks
✔️